packages/ui-kit/src/components/DataGrid/DataGrid.types.ts

```diff
@@ -20,22 +27,36 @@ export interface DataGridQueryProps extends Omit<QueryProps, 'metric'> {
    * @default 'DESC'
    * */
   sort?: Sort
+}
 
-  /**
-   * The number of rows to be returned when paging forward. It can be a number between 1 and 1,000.
-   * @default 50
-   * */
-  first?: number
-
-  /**
-   * The number of rows to be returned when paging forward. It can be a number between 1 and 1,000.
-   * @default 50
-   * */
-  last?: number
+export interface RowProps {
+  styles: CSSStyleSheet
+}
 
-  /** The cursor to use when paging forward. */
-  after?: string
+export interface DataGridProps extends DataComponentProps<'div'> {
+  query?: DataGridQueryProps
+  resizable?: boolean
+  /** Props to be applied to the `table` element */
+  tableProps?: React.HTMLAttributes<HTMLTableElement>
+  /** Props to be applied to the `td` elements */
+  cellProps?: React.HTMLAttributes<HTMLTableCellElement>
```

Comment:
Why do we need these? Our main approach to style the components is through the css, e.g. by passing a root level class:
```tsx
<DataGrid className={styles.myDataGrid} />
```
```scss
.myDataGrid {
  table {
    font-weight: bold;
    td {
      font-style: italic;
    }
  }
}
```

Reply:
This is in the RFC, I think there are good reasons to keep this approach rather than relying only on CSS selectors, lmk if you'd like to discuss further

Reply:
Following this pattern we ended up with a lot of outdated APIs. We should only introduce new props where it's really needed. There is a clear way for a user to control component styles (via CSS), we don't need to introduce additional props for that. So, let's keep things simple and remove `tableProps` and `cellProps` from the first version. We can introduce them later once we get a real use case for it.